File: spring-java/src/main/java/com/bsthun/w/chxpos/apidemo/spring/MvcConfigurer.java
package com.bsthun.w.chxpos.apidemo.spring;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@EnableWebMvc
public class MvcConfigurer implements WebMvcConfigurer {
	
	@Override
	public void addCorsMappings(CorsRegistry registry) {
		registry.addMapping("/**").allowedOrigins("http://localhost:3000", "https://dp-chxpos.bsthun.com");
	}
}

File: spring/src/main/kotlin/com/bsthun/w/chxpos/apidemo/beans/MvcConfigurer.kt
package com.bsthun.w.chxpos.apidemo.beans

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.EnableWebMvc
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import org.springframework.web.servlet.config.annotation.CorsRegistry

@Configuration
@EnableWebMvc
class MvcConfigurer : WebMvcConfigurer {
	override fun addCorsMappings(registry: CorsRegistry) {
		registry
			.addMapping("/**")
			.allowedOrigins("http://localhost:3000", "https://dp-chxpos.bsthun.com")
			.allowedMethods("GET", "POST", "PUT", "PATCH", "DELETE")
			.allowCredentials(true)
	}
}